### PR TITLE
Allow "inventory" actions to be supplied by a Module/Addon.

### LINKED ISSUE #5121: Bug: IPluginPaths missing inventory plugins
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior:

```
# Warning: ModulesManager : Expected keys from `get_plugin_paths` are "publish", "create", "load", "actions". Module: "bumpybox_addon" - got key "inventory"
```

No plugins can be registered for the inventory tool. Using `actions` does not work.

### Expected Behavior:

_No response_

### Version

3.15.10-nightly.2

### What platform you are running OpenPype on?

Windows

### Steps To Reproduce:

Add a module or addon that registers an inventory action. For example:

```python
"""Addon definition is located here.

Import of python packages that may not be available should not be imported
in global space here until are required or used.
- Qt related imports
- imports of Python 3 packages
    - we still support Python 2 hosts where addon definition should available
"""

import os

from openpype.modules import (
    OpenPypeAddOn,
    IPluginPaths,
    ITrayAction
)


class BumpyboxAddon(OpenPypeAddOn, IPluginPaths, ITrayAction):
    """This Addon has defined its settings and interface.

    This example has system settings with an enabled option. And use
    few other interfaces:
    - `IPluginPaths` to define custom plugin paths
    - `ITrayAction` to be shown in tray tool
    """
    label = "Bumpybox Addon"
    name = "bumpybox_addon"

    def tray_init(self):
        """Implementation of abstract method for `ITrayAction`.

        We're definitely  in tray tool so we can pre create dialog.
        """
        pass

    def on_action_trigger(self):
        """Implementation of abstract method for `ITrayAction`."""
        pass

    def get_plugin_paths(self):
        """Implementation of abstract method for `IPluginPaths`."""
        path = os.path.join(
            os.path.dirname(os.path.abspath(__file__)), "plugins"
        )
        return {
            "publish": [path],
            "inventory": [path]
        }
```

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the bug report.

### Relevant log output:

_No response_

### Additional context:

_No response_

[cuID:[OP-6158](https://app.clickup.com/t/861mwy7c9)]

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Adds "inventory" as a possible key to the plugin paths to be returned from a module.

## Additional info

Attempts to resolve #5121
It's untested :)

## Testing notes:

See steps to reproduce in  #5121 - it should now work

### CODE REVIEW

File: openpype/modules/interfaces.py [hunk 1]
```diff
@@ -397,8 +412,8 @@ def set_service_idle_icon(self):
 class ISettingsChangeListener(OpenPypeInterface):
     """Module has plugin paths to return.
 
-    Expected result is dictionary with keys "publish", "create", "load" or
-    "actions" and values as list or string.
+    Expected result is dictionary with keys "publish", "create", "load",
+    "actions" or "inventory" and values as list or string.
     {
```
Comment (BigRoy) [inline openpype/modules/interfaces.py:417]: I added it to the docstring here, but I suspect this docstring was already wrong to begin with hehe.

Comment (iLLiCiTiT) [reply openpype/modules/interfaces.py:417]: Yes, looks like the docstring was copied from `IPluginPath`.

Comment (BigRoy) [reply openpype/modules/interfaces.py:417]: Any decent replacement docstring you could write up for `ISettingsChangeListener`? :) I can include it here if that makes sense.

Comment (iLLiCiTiT) [reply openpype/modules/interfaces.py:417]: Not decent at all, but don't have time for more.
```
"""Module want to catch settings changes.

Only changes that happened in current process are propagated. Changes made in other processes, or machines won't  trigger the callbacks.
```

**Review by Minkiu — APPROVED** [review]
Haven't tested it, but code looks good.

**Review by tokejepsen — APPROVED** [review]
Tested in Maya 2023 successfully.

**Pull request merged**